src/Google.Storage.V1/StorageClientImpl.ListBuckets.cs

```diff
@@ -25,29 +25,30 @@ namespace Google.Storage.V1
     // ListBuckets methods on StorageClient
     public sealed partial class StorageClientImpl : StorageClient
     {
-        private static readonly PageStreamer<Bucket, BucketsResource.ListRequest, Buckets, string> s_bucketPageStreamer =
-            new PageStreamer<Bucket, BucketsResource.ListRequest, Buckets, string>(
-                (request, token) => request.PageToken = token,
-                buckets => buckets.NextPageToken,
-                buckets => buckets.Items);
+        private sealed class BucketPageManager : IPageManager<BucketsResource.ListRequest, Buckets, Bucket>
+        {
+            internal static readonly BucketPageManager Instance = new BucketPageManager();
```

Comment:
The methods are public because they're implementing an interface. The field *could* be public, but it doesn't make much difference when it's a private class anyway - I'd prefer to keep it as restrictive as possible though.